Objects/complexobject.c

```diff
@@ -887,22 +892,15 @@ complex_subtype_from_string(PyTypeObject *type, PyObject *v)
     PyObject *s_buffer = NULL, *result = NULL;
     Py_ssize_t len;
 
-    if (PyUnicode_Check(v)) {
-        s_buffer = _PyUnicode_TransformDecimalAndSpaceToASCII(v);
-        if (s_buffer == NULL) {
-            return NULL;
-        }
-        assert(PyUnicode_IS_ASCII(s_buffer));
-        /* Simply get a pointer to existing ASCII characters. */
-        s = PyUnicode_AsUTF8AndSize(s_buffer, &len);
-        assert(s != NULL);
-    }
-    else {
-        PyErr_Format(PyExc_TypeError,
-            "complex() argument must be a string or a number, not %T",
-            v);
+    assert(PyUnicode_Check(v));
```

Comment:
Can we actually add a comment on why this assertion is valid? now we know but in the future maybe we could forget? (namely pick what Pieter mentioned beforehand?)

Reply:
I think that function name is descriptive enough.